pandas/tests/indexes/test_base.py

```diff
@@ -1249,10 +1249,9 @@ def test_get_indexer_numeric_index_boolean_target(self, method, idx_class):
         if method == "get_indexer":
             tm.assert_numpy_array_equal(result, expected)
         else:
-            expected = np.array([-1, -1, -1, -1], dtype=np.intp)
-
+            missing = np.arange(3, dtype=np.intp)
             tm.assert_numpy_array_equal(result[0], expected)
-            tm.assert_numpy_array_equal(result[1], expected)
+            tm.assert_numpy_array_equal(result[1], missing)
```

Comment:
What's the reason for changing the behaviour here? 
Did you consider this a bug?

Reply:
Yes.  As mentioned in the OP, as far as i can tell it isnt actually affecting anything